cmd/nvidia-dra-controller/imex.go

```diff
@@ -99,29 +99,37 @@ func StartIMEXManager(ctx context.Context, config *Config) (*ImexManager, error)
 
 // manageResourceSlices reacts to added and removed IMEX domains and triggers the creation / removal of resource slices accordingly.
 func (m *ImexManager) manageResourceSlices(ctx context.Context, owner resourceslice.Owner, addedDomainsCh <-chan string, removedDomainsCh <-chan string) error {
-	driverResources := resourceslice.DriverResources{}
+	driverResources := resourceslice.DriverResources{
+		Pools: make(map[string]resourceslice.Pool),
+	}
 	controller, err := resourceslice.StartController(ctx, m.clientset, DriverName, owner, &driverResources)
 	if err != nil {
 		return fmt.Errorf("error starting resource slice controller: %w", err)
 	}
 
+	imexChannelOffsets := make(map[string]map[string]int)
+
 	m.waitGroup.Add(1)
 	go func() {
 		defer m.waitGroup.Done()
 		for {
 			select {
 			case addedDomain := <-addedDomainsCh:
 				klog.Infof("Adding channels for new IMEX domain: %v", addedDomain)
-				newDriverResources := DriverResources(driverResources).DeepCopy()
-				newDriverResources.Pools[addedDomain] = generateImexChannelPool(addedDomain, ImexChannelLimit)
-				controller.Update(&newDriverResources)
-				driverResources = newDriverResources
+				if err := calculateImexChannelLimit(addedDomain, &imexChannelOffsets); err != nil {
+					klog.Error(err)
+					return
+				}
```

Comment:
Do this before the log line as the very first thing in this function -- also wrap the returned error in a meanngful local string.